File: cmd/main/testdata/test_samples.go
package main

import (
	"fmt"
	"strings"
)

// SimpleFunction is a basic function for testing the parser
func SimpleFunction() {
	fmt.Println("Hello, world!")
}

// FunctionWithParams demonstrates a function with parameters
func FunctionWithParams(name string, age int) string {
	return fmt.Sprintf("Hello, %s! You are %d years old.", name, age)
}

// StructWithMethods demonstrates a struct with methods
type StructWithMethods struct {
	Name string
	Age  int
}

// GetInfo is a method on StructWithMethods
func (s *StructWithMethods) GetInfo() string {
	return fmt.Sprintf("%s is %d years old", s.Name, s.Age)
}

// UpdateAge updates the age and calls other functions
func (s *StructWithMethods) UpdateAge(newAge int) {
	s.Age = newAge
	SimpleFunction()
	FunctionWithParams(s.Name, s.Age)
	s.privateHelper()
}

// privateHelper is a private helper method
func (s *StructWithMethods) privateHelper() {
	strings.ToUpper(s.Name)
}

func main() {
	person := &StructWithMethods{Name: "Alice", Age: 30}
	fmt.Println(person.GetInfo())
	person.UpdateAge(31)
}

File: cmd/main/testdata/test_samples.cpp
// C++ test file for parser testing

#include <iostream>
#include <string>

// Simple function
void simpleFunction() {
    std::cout << "Hello, world!" << std::endl;
}

// Function with parameters
std::string functionWithParams(const std::string& name, int age) {
    return "Hello, " + name + "! You are " + std::to_string(age) + " years old.";
}

// Class with methods
class ClassWithMethods {
private:
    std::string name;
    int age;
    
    // Private helper method
    std::string privateHelper() {
        std::string result = name;
        for (auto& c : result) {
            c = std::toupper(c);
        }
        return result;
    }
    
public:
    ClassWithMethods(const std::string& name, int age) : name(name), age(age) {}
    
    std::string getInfo() {
        return name + " is " + std::to_string(age) + " years old";
    }
    
    void updateAge(int newAge) {
        age = newAge;
        simpleFunction();
        functionWithParams(name, age);
        privateHelper();
    }
};

// Main function
int main() {
    ClassWithMethods person("Alice", 30);
    std::cout << person.getInfo() << std::endl;
    person.updateAge(31);
    return 0;
}